test/sharness/lib/test-lib.sh

```diff
@@ -353,3 +355,38 @@ test_check_peerid() {
 	}
 }
 
+# A workaround for https://github.com/ipfs/go-ipfs/issues/1941 ‘sharness suite:
+# ipfs name publish: “Error: failed to find any peer in table”’.  Certain
+# commands fail whenever the daemon is running with no peers.
+# test_expect_success_1941 runs test_expect_failure if that is the case and
+# test_expect_success otherwise.  As soon as #1941 is fixed, all invocations of
+# test_expect_success_1941 can be replaced with test_expect_success and this
+# function can be removed.
```

Comment:
idk how i feel about this. could hide problems. right now we just run things again. 

i'd be :-1: on this